### PR TITLE
Cody: Add submit button to edit dialog for parity with chat input box

### LINKED ISSUE #51740: Cody Bug: Edit field behaves different
RE: https://sourcegraph.slack.com/archives/C052G9Y5Y8H/p1683741607462599

filed by erik:

> UI feedback: It’s not very obvious how to persist the change I made. There’s no paperplane button like there is on the regular text field. it seems to be pressing enter but I got confused at first. then, the change is only shown in the UI once the request goes through, not immediately

Image

### PR DESCRIPTION
**Pull request by SuperAuguste**
Closes #51740

## Test plan

Tested web and VSCode locally.

### CODE REVIEW

**Comment by sourcegraph-buildkite** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.01% (+1.51 kb) | 0.01% (+1.51 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/auguste/cody-edit-submit/compare-report.html;tab=live_object) for a full comparison between the commits 872506b and 2f9fb8e or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by abeatrix — CHANGES_REQUESTED** [review]
There is a styling issue that is causing the view jumps when trying to open the context files list 😅 

### Current released version:
current

### In your PR:
pr

File: client/cody-ui/src/chat/TranscriptItem.tsx [hunk 1]
```diff
@@ -62,39 +70,51 @@ export const TranscriptItem: React.FunctionComponent<
...
-                    }
+        TextArea && beingEdited && editButtonOnSubmit && SubmitButton ? (
+            <div className={styles.textAreaContainer}>
+                <TextArea
+                    className={classNames(styles.chatInput, chatInputClassName)}
+                    rows={5}
+                    value={formInput}
+                    autoFocus={true}
+                    required={true}
+                    onInput={({ target }) => {
```
Comment (abeatrix) [inline client/cody-ui/src/chat/TranscriptItem.tsx:86]:
```suggestion
                    onInput={e => setFormInput((e.target as HTMLInputElement).value)}
```
nit 😄

File: client/cody-ui/src/chat/TranscriptItem.tsx [hunk 2]
```diff
@@ -62,39 +70,51 @@ export const TranscriptItem: React.FunctionComponent<
...
-                        formInput &&
-                        formInput.trim()
-                    ) {
-                        event.preventDefault()
-                        event.stopPropagation()
+                        if (
+                            event.key === 'Enter' &&
+                            !event.shiftKey &&
+                            !event.nativeEvent.isComposing &&
+                            formInput &&
```
Comment (abeatrix) [inline client/cody-ui/src/chat/TranscriptItem.tsx:99]:
```suggestion

```
I think we can remove this and just use `formInput.trim()`

File: client/cody-ui/src/chat/TranscriptItem.tsx [hunk 3]
```diff
@@ -62,39 +70,51 @@ export const TranscriptItem: React.FunctionComponent<
...
+                            formInput.trim()
+                        ) {
+                            event.preventDefault()
+                            event.stopPropagation()
```
Comment (abeatrix) [inline client/cody-ui/src/chat/TranscriptItem.tsx:103]:
```suggestion
                            event.stopPropagation()
```
I know you didn't add this but I don't think we need it anymore haha 🙇‍♀️

**Review by abeatrix — APPROVED** [review]
👍 works great! Thank you for working on this!

**Pull request merged**